app/src/main/java/org/sagebionetworks/research/mpower/tracking/fragment/AddNoteFragment.java

```diff
@@ -0,0 +1,118 @@
+package org.sagebionetworks.research.mpower.tracking.fragment;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.view.OnApplyWindowInsetsListener;
+import android.support.v4.view.ViewCompat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+
+import org.sagebionetworks.research.mobile_ui.show_step.view.SystemWindowHelper;
+import org.sagebionetworks.research.mobile_ui.show_step.view.SystemWindowHelper.Direction;
+import org.sagebionetworks.research.mobile_ui.widget.ActionButton;
+import org.sagebionetworks.research.mpower.R;
+import org.sagebionetworks.research.mpower.tracking.model.TrackingItem;
+import org.sagebionetworks.research.mpower.tracking.view_model.configs.TrackingItemConfig;
+import org.sagebionetworks.research.mpower.tracking.view_model.logs.NoteLog;
+import org.sagebionetworks.research.mpower.tracking.view_model.logs.TrackingItemLog;
+import org.sagebionetworks.research.mpower.tracking.view_model.TrackingTaskViewModel;
+import org.sagebionetworks.research.presentation.model.interfaces.StepView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import butterknife.BindView;
+
+/**
+ * This fragment displays the add note screen to the user and allows the user to edit/add a new note. It is an
+ * invariant in this fragment that there is already a log in the view model for the tracking item provided as an
+ * argument and the client of this fragment is responsible for ensuring this is correct.
+ * @param <ConfigType> The type of TrackingItemConfig
+ * @param <LogType> The type of TrackingItemLog
+ * @param <ViewModelType> The type of view model.
+ */
+public class AddNoteFragment
```

Comment:
I feel like this fragment should be a re-usable title/text, textfield input fragment. The Symptom tracking screen would just get the result and use it.